app/forms/providers/partners/bank_statement_form.rb

```diff
@@ -0,0 +1,38 @@
+module Providers
+  module Partners
+    class BankStatementForm < Providers::BankStatementForm
+      def initialize(bank_statement_form_params)
+        super(bank_statement_form_params)
+        @attachment_type = "partner_bank_statement_evidence"
+        @attachment_type_capture = /^#{@attachment_type}_(\d+)$/
+      end
+
+    private
+
+      def any_bank_statements_or_draft?
+        legal_aid_application.attachments.partner_bank_statement_evidence.any? || draft?
+      end
+
+      # can be shared with v1 bank statement controller
+      def create_attachment(file)
+        attachment = legal_aid_application
+                        .attachments.create!(document: file,
+                                             attachment_type: "partner_bank_statement_evidence",
```

Comment:
nit: can we pass `@attachment_type` instead of the string  `partner_bank_statement_evidence`

Reply:
could add attribute reader to superclass and then transparently use in some places and explicitly in others,

```
# in superclass
attr_reader :attachment_type
...
# in sublcass
def initialize
@attachement_type = 'whatever'
...
end
```

use transparently
```...
     def create_attachment(file)
        attachment = legal_aid_application
                        .attachments.create!(document: file,
                                             attachment_type: ,
                                             ...

```
or explicitly
```
      def sequenced_attachment_name
        if legal_aid_application.attachments.partner_bank_statement_evidence.any?
          ...
        else
          attachment_type
        end
      end
```

Reply:
looks good!